src/common/actions/repository-input.js

```diff
@@ -52,3 +58,48 @@ export function verifyGitHubRepositoryError(error) {
     error: true
   };
 }
+
+export function createSnap(repository, location) {
+  return (dispatch) => {
+    if (repository) {
+      dispatch({
+        type: CREATE_SNAP,
+        payload: repository
+      });
+
+      return fetch(`${BASE_URL}/api/launchpad/snaps`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          repository_url: `https://github.com/${repository}`
+        }),
+        credentials: 'same-origin'
+      })
+        .then(checkStatus)
+        .then(response => {
+          return response.json().then(result => {
+            if (result.status !== 'success' ||
+                result.payload.code !== 'snap-created') {
+              const error = new Error(response.statusText);
+              error.response = response;
+              throw error;
+            }
+            const startingUrl = `${BASE_URL}/${repository}/builds`;
```

Comment:
I think there's probably no point at the moment, since at this point we have to redirect through OpenID so we're about to have to reload the store from scratch anyway.